front/components/agent_builder/capabilities/knowledge/utils.ts

```diff
@@ -0,0 +1,124 @@
+import type { ContentNodesViewType } from "@dust-tt/client";
+import {
+  ActionIncludeIcon,
+  ActionScanIcon,
+  MagnifyingGlassIcon,
+  TableIcon,
+} from "@dust-tt/sparkle";
+
+import {
+  DESCRIPTION_MAX_LENGTH,
+  type AgentBuilderAction,
+  type KnowledgeServerName,
+} from "@app/components/agent_builder/types";
+
+export interface CapabilityConfig {
+  name: KnowledgeServerName;
+  title: string;
+  description: string;
+  icon: React.ComponentType;
+  viewType: ContentNodesViewType;
+  actionType: AgentBuilderAction["type"];
+  actionName: string;
+  configPageTitle: string;
+  configPageDescription: string;
+  hasTimeFrame: boolean;
+  hasJsonSchema: boolean;
+  descriptionConfig: {
+    title: string;
+    description: string;
+    placeholder: string;
+    helpText?: string;
+    maxLength?: number;
+  };
+}
+
+export const CAPABILITY_CONFIGS: Record<KnowledgeServerName, CapabilityConfig> =
```

Comment:
I generated this based on the current implementation of AB v2, but I now wonder if we should reuse `getMCPServerRequirements` and get requirements from selected MCP server view? 🤔